### PR TITLE
Add C Serialization Tests for #17217

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

#17217 fix went in but didn't contain a verifying test

#### Summary of Changes

Add test

Fixes #

### CODE REVIEW

File: programs/bpf/c/src/ser/ser.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    return ERROR_INVALID_ARGUMENT;
+  }
+
+  char ka_data[] = {1, 2, 3};
+  SolPubkey ka_owner;
+  sol_assert(params.ka_num == 2);
+  for (int i = 0; i < 2; i++) {
+    sol_assert(*params.ka[i].lamports == 42);
+    sol_assert(!sol_memcmp(params.ka[i].data, ka_data, 4));
+    sol_assert(SolPubkey_same(params.ka[i].owner, &ka_owner));
```
Comment (dmakarov) [inline programs/bpf/c/src/ser/ser.c:23]: This test seems to fail consistently at this line.  Why should it work if local struct `ka_owner` is not initialized?  What am I missing?..

**Comment by jackcmay** [comment]
Test is probably assuming zero initialized memory

**Comment by dmakarov** [comment]
but the `params.ka[i].owner` is not zero.

**Comment by jackcmay** [comment]
Fixed in: https://github.com/solana-labs/solana/pull/17310

**Pull request merged**